### ISSUE TITLE
Invalid requestBody schema with $ref (OAS 3.0)

### ISSUE THREAD

**Issue by ViktorPegy** [drwpow/openapi-typescript#493]
openapi-typescript version: 3.0.1 (npx)

example `openapi.yaml`:
```yaml
openapi: 3.0.2
components:
  requestBodies:
    TestBody:
      content:
        application/json:
          schema:
            $ref: "#/components/schemas/Test"
  schemas:
    Test:
      type: object
      properties:
        name:
          type: string
paths:
  /test/:
    post:
      summary: test
      operationId: test-op-id
      requestBody:
        $ref: "#/components/requestBodies/TestBody"
```

generated code:
```typescript
export interface paths {
  "/test/": {
    post: operations["test-op-id"];
  };
}

export interface components {
  schemas: {
    Test: {
      name?: string;
    };
  };
  requestBodies: {
    TestBody: { [key: string]: any }; // <---- invalid
  };
}

export interface operations {
  "test-op-id": {
    requestBody: components["requestBodies"]["TestBody"];
  };
}
```

`TestBody` must be like:
```typescript
requestBodies: {
  TestBody: {
    content: {
      "application/json": components["schemas"]["Test"];
    };
  };
};
```

**Comment by gr2m** [comment]
could you start a pull request with a failing test? You could add your schema to `tests/v3/specs`, then generated the output as described in `CONTIRBUTING.md`, then adjust the generated output in `tests/v3/expected` to what it should be. We can continue the discussion there, it will make it easier to move forward

**Comment by drwpow** [comment]
I believe this was fixed with #474 and released in `3.0.2` yesterday. Can you confirm?

**Comment by ViktorPegy** [comment]
Yes, it's looks good
